crates/resvg/src/image.rs

```diff
@@ -87,7 +87,25 @@ mod raster_images {
 
         let options = DecoderOptions::default().jpeg_set_out_colorspace(ColorSpace::RGBA);
         let mut decoder = zune_jpeg::JpegDecoder::new_with_options(data, options);
-        let img_data = decoder.decode().ok()?;
+        decoder.decode_headers().ok()?;
+        let output_cs = decoder.get_output_colorspace()?;
+
+        let img_data = {
+            let data = decoder.decode().ok()?;
+            match output_cs {
+                ColorSpace::RGBA => data,
+                // `set_output_color_space` is not guaranteed to actually always set the output space
+                // to RGBA (see https://docs.rs/zune-core/latest/zune_core/options/struct.DecoderOptions.html#method.jpeg_set_out_colorspace).
```

Comment:
```suggestion
                // to RGBA (its docs say "we do not guarantee the decoder can convert to all colorspaces").
```
I don't really like to link directly to docs.rs in Rust code, because the same information is already available to you locally as you're reading this code (as `jpeg_set_out_colorspace` is used 10 lines above)